stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/search/SearchViewModel.kt

```diff
@@ -5,57 +5,116 @@ import androidx.lifecycle.MutableLiveData
+            isLoading = false,
+            isError = false,
+            canLoadMore = true,
+        )
+    }
+
+    private val scope = CoroutineScope(DispatcherProvider.Main + SupervisorJob())
+
+    override fun onCleared() {
+        super.onCleared()
+        scope.cancel()
+    }
+
+    private val _state: MutableLiveData<State> = MutableLiveData(INITIAL_STATE)
+    public val state: LiveData<State> = _state
+
+    private val logger = ChatLogger.get("SearchViewModel")
 
     public fun setQuery(query: String) {
-        fetchDummyResults(query)
+        scope.launch {
+            _state.value = State(
+                query = query,
+                results = emptyList(),
+                isLoading = true,
+                isError = false,
+                canLoadMore = true,
+            )
+            fetchServerResults()
+        }
     }
 
-    private fun fetchDummyResults(query: String) {
-        _results.value = List(5) {
-            Message(
-                id = "dummy-id-$it",
-                cid = "messaging:placeholder",
-                user = User().apply {
-                    name = "John Doe"
-                    image = "https://randomuser.me/api/portraits/men/0.jpg"
-                },
-                createdAt = Date(2020, 3, 15, 18, 11),
-                text = "Message with \"$query\" in it",
+    public fun loadMore() {
+        scope.launch {
+            val currentState = _state.value!!
+
+            if (!currentState.canLoadMore) {
+                logger.logD("No more messages to load")
+                return@launch
+            }
+            if (currentState.isLoading) {
+                logger.logD("Already loading")
+                return@launch
+            }
+
+            _state.value = currentState.copy(
+                isLoading = true,
             )
+            fetchServerResults()
         }
     }
 
-    private fun fetchServerResults(query: String) {
-        // TODO update these filters
-        val channelFilter = Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()!!.id))
-        val messageFilter = Filters.autocomplete("text", query)
+    private suspend fun fetchServerResults() {
+        val currentState = _state.value!!
+        val currentUser = requireNotNull(ChatClient.instance().getCurrentUser())
+        val channelFilter = Filters.`in`("members", listOf(currentUser.id))
+        val messageFilter = Filters.autocomplete("text", currentState.query)
 
-        // TODO add paging
         val request = SearchMessagesRequest(
-            offset = 0,
-            limit = 30,
+            offset = currentState.results.size,
+            limit = QUERY_LIMIT,
             channelFilter = channelFilter,
             messageFilter = messageFilter,
         )
-        ChatClient.instance()
+
+        logger.logD("Searching (offset: ${request.offset}, limit: ${request.limit}, query: ${currentState.query})")
+
+        val result = ChatClient.instance()
```

Comment:
We'll need support this in offline SDKs if we want to use the SSoT approach in the future